Library/Homebrew/cmd/update.sh

```diff
@@ -390,29 +390,24 @@ EOS
     fi
   fi
 
-  if [[ -f "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-core/.git/shallow" ]]
-  then
-    odie <<EOS
-homebrew-core is a shallow clone. To \`brew update\` first run:
-  git -C "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-core" fetch --unshallow
-This restriction has been made on GitHub's request because updating shallow
-clones is an extremely expensive operation due to the tree layout and traffic of
-Homebrew/homebrew-core. We don't do this for you automatically to avoid
-repeatedly performing an expensive unshallow operation in CI systems (which
-should instead be fixed to not use shallow clones). Sorry for the inconvenience!
-EOS
-  fi
+  [[ -f "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-core/.git/shallow" ]] && HOMEBREW_CORE_SHALLOW=1
+  [[ -f "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-cask/.git/shallow" ]] && HOMEBREW_CASK_SHALLOW=1
 
-  if [[ -f "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-cask/.git/shallow" ]]
+  if [[ -n $HOMEBREW_CORE_SHALLOW || -n $HOMEBREW_CASK_SHALLOW ]]
   then
     odie <<EOS
-homebrew-cask is a shallow clone. To \`brew update\` first run:
-  git -C "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-cask" fetch --unshallow
-This restriction has been made on GitHub's request because updating shallow
-clones is an extremely expensive operation due to the tree layout and traffic of
-Homebrew/homebrew-cask. We don't do this for you automatically to avoid
-repeatedly performing an expensive unshallow operation in CI systems (which
-should instead be fixed to not use shallow clones). Sorry for the inconvenience!
+  ${HOMEBREW_CORE_SHALLOW:+
+    homebrew-core is a shallow clone.}${HOMEBREW_CASK_SHALLOW:+
+    homebrew-cask is a shallow clone.}
+  To \`brew update\`, first run:${HOMEBREW_CORE_SHALLOW:+
+    git -C "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-core" fetch --unshallow}${HOMEBREW_CASK_SHALLOW:+
+    git -C "$HOMEBREW_LIBRARY/Taps/homebrew/homebrew-cask" fetch --unshallow}
+  This restriction has been made on GitHub's request because updating shallow
+  clones is an extremely expensive operation due to the tree layout and traffic of
+  Homebrew/homebrew-core and Homebrew/homebrew-cask. We don't do this for you
+  automatically to avoid repeatedly performing an expensive unshallow operation in
+  CI systems (which should instead be fixed to not use shallow clones). Sorry for
+  the inconvenience!
```

Comment:
I think we can remove couple of spaces from the beginning of every line here

Reply:
Can you post what you want the actual output to look like? Going back and forth adding and removing spaces is giving me flashbacks to sitting with my designer and moving things by 1px "to see if that makes a difference"

Reply:
IMO, the indentation makes this message stand out more and is desirable.

Reply:
I agree with @rgant!